### PR TITLE
06732 - ConsensusImpl Supports Noncontiguous NodeId

### LINKED ISSUE #6732: Noncontiguous NodeId - ConsensusImpl and Related Files
Use NodeId API from AddressBook

For each batch, these are the changes that need to be made. 
- NodeId - All methods returning a long node id now return NodeId
- NodeId - All methods consuming a long node id now consume NodeId
- NodeId - Change usage of int node ids as index of arrays/lists to  Map<NodeId, ?>
- NodeId - Complete translation of all other usage of int/long node ids to NodeId
- Ensure unit tests work with noncontiguous NodeId

Files:
- ConsensusImpl
- QuorumTriggerAction
- SyncManagerImpl.getNeighborsToCall()
- StandardGraphGenerator.buildAddressBook()
- FallenBehindManager
- ConsensusTestDefinitions - FastConsensusTest - dependent on RandomAddressBookGenerator being sequential
- EventFlowTests - SwirldStateFlowTests - depend on ConsensusImpl with Sequential Ids. 
- SimpleEventGenerator - nextEvent - generates a random node id based on number of nodes. 
- ChatterInstance - NodeBuilder - ChatterSimulationTests - All testing dependent on contiguous node Ids. 
- StandardGraphGenerator - does this need to use NodeId instead of indexes? 
- Any other that have been delayed to avoid merge conflicts.

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:

Fixes #6732

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 1]
```diff
@@ -1435,28 +1443,23 @@ private EventImpl firstWitnessS(EventImpl x) {
...
+    private EventImpl stronglySeeS1(EventImpl x, int m) {
         return stronglySeeP(firstWitnessS(x), m);
     }
 
     /**
-     * The first witness in round r that is a self-ancestor of x, where r is the round of the last event by m
-     * that is
-     * seen by x (function from SWIRLDS-TR-2020-01).
-     * This result is not memoized.
+     * The first witness in round r that is a self-ancestor of x, where r is the round of the last event by m that is
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1454]: `by m` -> `by the member at index m`

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java [hunk 2]
```diff
@@ -42,9 +42,11 @@
 import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.Objects;
 import java.util.Random;
 import java.util.function.Consumer;
 import java.util.function.Predicate;
+import org.checkerframework.checker.nullness.qual.NonNull;
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java:49]: Why are these tests (and others) including a different annotation than normal?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java [hunk 3]
```diff
@@ -687,13 +688,15 @@ void testSendExpiredEvents() throws Exception {
     @ParameterizedTest
     @MethodSource({"tenNodeGraphParams", "tenNodeBigGraphParams", "tipExpiresBreakingSeed"})
     void tipExpiresAfterPhase1(final SyncTestParams params) throws Exception {
-        final long creatorIdToExpire = 0;
         final SyncTestExecutor executor = new SyncTestExecutor(params);
         final AtomicLong maxGen = new AtomicLong(EventConstants.GENERATION_UNDEFINED);
 
+        final NodeId creatorIdToExpire = executor.getAddressBook().getNodeId(0);
+        final int creatorIndex = executor.getAddressBook().getIndexOfNodeId(creatorIdToExpire);
```
Comment (alittley) [inline platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java:695]: `creatorIndex` is just 0, right?

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java:49]: huh, not the findbugs?   It must need to be added to the module as a dependency?   I'll check.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java (hunk 3 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java:695]: lol.  yeah I guess it is.

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/sync/SyncTests.java:695]: The `creatorIdToExpire` is used elsewhere.   I swapped the order of initializing the variables.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1454]: updated.

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java:49]: I am not able to change the the import statement to use `import edu.umd.cs.findbugs.annotations.NonNull`.  IntelliJ asks me to add the library to the class path and doesn't do anything when I click on the option.

Comment (alittley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java:49]: weird, it's not obvious to me what's going on. This other library seems to have been in use in the module for a while, though, so I suppose it's fine

Comment (hendrikebbers) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java:49]: I am not able to change the the import statement to use import edu.umd.cs.findbugs.annotations.NonNull.  <- WHY? Do you mean that IntelliJ directly refactors it back to the wrong annotation?

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/eventflow/EventFlowWrapper.java]: fixed.   I had to add `testCompileOnly(...spotbugs...)` in the build as a dependency.   I went ahead and updated all files with the change.

File: platform-sdk/swirlds-platform-core/build.gradle.kts [hunk 4]
```diff
@@ -15,41 +15,43 @@
...
-  compileOnly(libs.spotbugs.annotations)
-  runtimeOnly(project(":swirlds-config-impl"))
+    // Individual Dependencies
+    implementation(project(":swirlds-base"))
+    api(project(":swirlds-fchashmap"))
+    api(project(":swirlds-fcqueue"))
+    api(project(":swirlds-jasperdb"))
+    api(project(":swirlds-cli"))
+    api(project(":swirlds-base"))
+    testImplementation("org.jetbrains:annotations:23.0.0")
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/build.gradle.kts:34]: I'm assuming jetbrains added this automatically - shouldn't be necessary

File: platform-sdk/swirlds-platform-core/build.gradle.kts [hunk 5]
```diff
@@ -15,41 +15,43 @@
...
+    compileOnly(libs.spotbugs.annotations)
+    testCompileOnly(libs.spotbugs.annotations)
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/build.gradle.kts:36]: This should go down in the `Test Dependencies` section (applies to all `build.gradle` files in this PR)

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 6]
```diff
@@ -1111,26 +1115,26 @@ private long parentRound(EventImpl x) {
...
      * @param x
      * 		the event being queried
      * @param m
-     * 		the member ID of the creator
-     * @return the last event created by m that is an ancestor of x, or null if none
+     * 		the index of the member ID of the creator
+     * @return the last event created by member at index m that is an ancestor of x, or null if none
      */
-    private EventImpl lastSee(EventImpl x, long m) {
+    private EventImpl lastSee(EventImpl x, int m) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1129]: I know this is old code, but maybe we can start adding this kind of stuff as we make changes.

```suggestion
    private EventImpl lastSee(@Nullable final EventImpl x, final int m) {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 7]
```diff
@@ -1156,27 +1161,28 @@ private EventImpl lastSee(EventImpl x, long m) {
...
-     * 		the creator of y, the event seen
+     * 		the index of the creator of y, the event seen
      * @param m2
-     * 		the creator of z, the intermediate event through which x sees y
-     * @return the event y that is created by m and seen by x through an event by m2
+     * 		the index of the creator of z, the intermediate event through which x sees y
+     * @return the event y that is created by the member at index m and seen by x through an event by the member at index m2
      */
-    private EventImpl seeThru(EventImpl x, long m, long m2) {
+    private EventImpl seeThru(EventImpl x, int m, int m2) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1180]:
```suggestion
    private EventImpl seeThru(@Nullable final EventImpl x, final int m, final int m2) {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 8]
```diff
@@ -1193,17 +1199,17 @@ private EventImpl seeThru(EventImpl x, long m, long m2) {
      * @param x
      * 		the event being queried
      * @param m
-     * 		the member ID of the creator
-     * @return witness created by m in the parent round of x that x strongly sees, or null if none
+     * 		the index of the member ID of the creator
+     * @return witness created by the member at index m in the parent round of x that x strongly sees, or null if none
      */
-    private EventImpl stronglySeeP(EventImpl x, long m) {
+    private EventImpl stronglySeeP(final EventImpl x, final int m) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1205]:
```suggestion
    private EventImpl stronglySeeP(@Nullable final EventImpl x, final int m) {
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 9]
```diff
@@ -1435,28 +1443,23 @@ private EventImpl firstWitnessS(EventImpl x) {
...
-     * @return firstSelfWitnessS(lastSee ( x, m)), which is the first witness in round r that is a
-     * 		self-ancestor
-     * 		of x, where r is the round of the last event by m that is seen by x, or null if none
+     * @param x the event being queried
+     * @param m the index of the member ID of the creator
+     * @return firstSelfWitnessS(lastSee ( x, m)), which is the first witness in round r that is a self-ancestor of x,
+     * where r is the round of the last event by the member at index m that is seen by x, or null if none
      */
-    private EventImpl firstSee(EventImpl x, long m) {
+    private EventImpl firstSee(EventImpl x, int m) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1462]:
```suggestion
    private EventImpl firstSee(@Nullable final EventImpl x, final int m) {
```

File: platform-sdk/swirlds-platform-core/build.gradle.kts (hunk 5 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/build.gradle.kts:36]: moved

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/build.gradle.kts]: ah!

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/build.gradle.kts]: should be fixed in the 3 other locations now.  sorry for not fully reading your comment.

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/build.gradle.kts (hunk 5 above)
Comment (alittley) [reply platform-sdk/swirlds-platform-core/build.gradle.kts]: No worries 👍

**Review by nathanklick — APPROVED** [review]
LGTM, but you may want to run spotless apply.

**Pull request merged**